pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -903,6 +918,22 @@ func getFailingSyncSets(syncStatuses []hiveintv1alpha1.SyncStatus) []string {
 	return failures
 }
 
+func (r *ReconcileClusterSync) setFirstSyncSetsSuccessTime(syncStatuses []hiveintv1alpha1.SyncStatus, cd *hivev1.ClusterDeployment, clusterSync *hiveintv1alpha1.ClusterSync, logger log.FieldLogger) {
+	lastSuccessTime := &metav1.Time{}
+	for _, status := range syncStatuses {
+		if status.FirstSuccessTime == nil {
+			return
+		}
+		if status.FirstSuccessTime.Time.After(lastSuccessTime.Time) {
+			lastSuccessTime = status.FirstSuccessTime
+		}
+	}
+	clusterSync.Status.FirstSyncSetsSuccessTime = lastSuccessTime
+	allSyncSetsAppliedDuration := lastSuccessTime.Time.Sub(cd.Status.InstalledTimestamp.Time)
+	metricTimeToApplySyncSets.Observe(float64(allSyncSetsAppliedDuration.Seconds()))
```

Comment:
Feels like this could just be clusterSync.Status.FirstSyncSetsSuccessTime := time.Now(), i.e. the first time we observed that everything was ok.

Reply:
I like tying this to the `lastSuccessTime` of a `syncStatus` so that it is predictable in a test.

Reply:
Fair point, sounds good to me.